### DISCUSSION TITLE
How to use Azure Application Insights

### DISCUSSION THREAD

**Discussion by DiomedesDominguez** [OrchardCMS/OrchardCore#7923] [Show and tell]
I was searching in all the issues (closed and opened) regarding the implementation of Azure Application Insights and I couldn't find any related one. The idea I have is that, as the same we can ```.AddAzureShellsConfiguration() ``` we can have ```.AddApplicationInsights() ``` and read from the ``` appsettings.json``` file the instrumentation key and URI.

This add a great value because if we use for example the Azure WebApp for containers right now we can't access the ```App_Data/logs/internal-nlog.txt``` file using SSH because the port isn't expose.

**Comment by Piedone** [comment]
We might update our [Orchard 1 Application Insights module](https://github.com/Lombiq/Orchard-Azure-Application-Insights) with a similar goal.

**Reply by Piedone** [reply]
We're working on creating a new module in some way, I'll publish it when ready.

**Reply by Piedone** [reply]
We've rewritten it for OC, check it out here: https://github.com/Lombiq/Orchard-Azure-Application-Insights

**Comment by DiomedesDominguez** [comment]
For now, this implementation works for me:
1) Download the source code from dev branch
2) Add the AI and the required settings.
3) Copy the required DLLs to a docker image or any other environment.

https://github.com/LinuxDominicana/OrchardCore/commit/a08adb43d1f18047e8d35b354445346148629ff3

![image](https://user-images.githubusercontent.com/4379316/101796826-b8f34200-3adf-11eb-84a6-aba2d62846e6.png)

**Comment by Skrypt** [comment]
Yeah this is a simple change. We should not provide this by default in the source code. And if you have a Nuget packages solution then it is on your own OrchardCore.Cms.Web project that this should be added. So, I think we should just produce some documentation in this case. A module could work too, but that would be really simple.
